diff --speed-large-files --minimal -Nru tmp/perl-5.10.1/Configure work/perl-5.10.1/Configure
--- tmp/perl-5.10.1/Configure	2009-08-18 21:03:53.000000000 +0200
+++ work/perl-5.10.1/Configure	2011-07-22 11:39:09.014556709 +0200
@@ -1308,6 +1308,7 @@
 
 : general looking path for locating libraries
 glibpth="/lib /usr/lib $xlibpth"
+glibpth="$glibpth /usr/lib/i386-linux-gnu /usr/lib/x86_64-linux-gnu"
 glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
