--- work/Filesys-DiskFree-0.06/DiskFree.pm	2006-05-18 20:26:20.000000000 +0200
+++ work/Filesys-DiskFree-0.06/DiskFree.pm.new	2006-06-16 13:02:20.000000000 +0200
@@ -228,6 +228,7 @@
 sub extract () {
     my $self=shift;
     my $device;
+
     if(@_) {
 	my $thingy=shift;
 	if(defined($self->{'DEVICES'}{$thingy})){
@@ -235,7 +236,9 @@
 	} else {
 	    return undef unless(defined($self->{'MOUNTS'}));
 	    while(not defined($self->{'MOUNTS'}{$thingy})){
+	      
 		return undef if($thingy eq '/');
+		-l $thingy and $thingy=readlink $thingy and next;
 		$thingy =~ s!/[^/]*?$!!  unless($thingy =~ s!/$!!);
 		$thingy = "/" unless($thingy =~ "/");
 	    }
