--- work/Filesys-DiskFree-0.06/DiskFree.pm	1998-10-24 00:30:01.000000000 +0200
+++ work/Filesys-DiskFree-0.06/DiskFree.pm.new	2006-03-06 17:10:37.000000000 +0100
@@ -109,7 +109,10 @@
     $cmd=$self->command() or
     	croak "No df command known for format ".$self->{'FORMAT'};
     open(HANDLE,"$cmd|") or croak("Cannot fork $!");
-    return $self->load(\*HANDLE);
+    if($self->load(\*HANDLE)){
+	close(HANDLE);
+	return 'true';
+    }
     close(HANDLE) or croak("Cannot df $!");
 }
 
