--- work/Getopt-Declare-1.11/lib/Getopt/Declare.pm	2004-02-03 21:44:27.000000000 +0100
+++ work/Getopt-Declare-1.11/lib/Getopt/Declare.pm.new	2006-02-17 18:27:45.000000000 +0100
@@ -1018,7 +1018,7 @@
 	my $filedate = localtime((stat $0)[9]);
 	if ($::VERSION) { print "\n\t$0: version $::VERSION  ($filedate)\n\n" }
 	else		{ print "\n\t$0: version dated $filedate\n\n" }
-	exit $_[1] if defined $_[1];
+	die $_[1] if defined $_[1];
 }
 
 sub usage
@@ -1131,7 +1131,7 @@
 	print $PAGER  "Options:\n" unless $decfirst;
 	print $PAGER  $usage;
 
-	exit $_[1] if defined $_[1];
+	die $_[1] if defined $_[1];
 }
 
 sub unused {
