--- work/Crypt-SSLeay-0.57/Makefile.PL.old	2009-08-25 16:16:30.000000000 +0200
+++ work/Crypt-SSLeay-0.57/Makefile.PL	2009-08-25 16:17:04.000000000 +0200
@@ -282,9 +282,7 @@
 these tests may fail (through no fault of the code).
 
 INFO
-    my $network_tests = prompt
-        "Do you want to run the live tests (y/N) ?",
-        'N';
+    my $network_tests = 'N';
     print OUT "network_tests ", ($network_tests =~ /y/i) ? 1 : 0, "\n";
     close OUT;
 }
