--- work/Authen-PAM-0.16/Makefile.PL.old	2009-02-10 11:42:16.000000000 +0100
+++ work/Authen-PAM-0.16/Makefile.PL	2009-02-10 11:42:42.000000000 +0100
@@ -46,7 +46,8 @@
 WriteMakefile(
     'NAME'	=> 'Authen::PAM',
     'VERSION_FROM' => 'PAM.pm',
-    'LIBS'	=> ['-lpam'],
+    'CCCDLFLAGS' => $ENV{'CFLAGS'},
+    'LIBS'	=> [$ENV{'LDFLAGS'}.' -lpam'],
     'INC'       => '-I.',          # Needed for PAM_config.h
     'CONFIGURE' => \&configure,
     'PREREQ_PM' => { POSIX => 0 }, # module dependenices
