diff --speed-large-files --minimal -Nru tmp/xrootd-3.0.4/config/GNUmake.rules.gcc work/xrootd-SHUTTLE/config/GNUmake.rules.gcc
--- tmp/xrootd-3.0.4/config/GNUmake.rules.gcc	2011-05-04 17:39:30.000000000 +0200
+++ work/xrootd-3.0.4/config/GNUmake.rules.gcc	2011-05-05 15:19:24.000000000 +0200
@@ -20,4 +20,4 @@
 TYPELDSO  += -m32
 endif
 
-TYPELIBS   =  -lnsl $(PTHREAD) -lrt -ldl -lc
+TYPELIBS   =  -lnsl $(PTHREAD) -lrt -ldl -lc -lpthread
diff --speed-large-files --minimal -Nru tmp/xrootd-3.0.4/config/GNUmake.rules.gccx8664 work/xrootd-3.0.4/config/GNUmake.rules.gccx8664
--- tmp/xrootd-3.0.4/config/GNUmake.rules.gccx8664	2011-05-04 17:39:30.000000000 +0200
+++ work/xrootd-3.0.4/config/GNUmake.rules.gccx8664	2011-05-05 15:19:56.000000000 +0200
@@ -16,4 +16,4 @@
 TYPELDSO   = -m64 -shared
 TYPESHLIB  = so
 
-TYPELIBS   =  -lnsl $(PTHREAD) -lrt -ldl -lc
+TYPELIBS   =  -lnsl $(PTHREAD) -lrt -ldl -lc -lpthread
