--- work/ApMon-CPP-2.0.5/ApMon.h	Wed Jul 13 13:27:16 2005
+++ work/ApMon-CPP-2.0.5/ApMon.h.new	Wed Jul 13 13:27:08 2005
@@ -54,10 +54,16 @@
 #include <pthread.h>
 #include <pwd.h>
 #include <grp.h>
+
+#ifdef __APPLE__
+#include <sys/param.h>
+#else
 #include <linux/param.h>
+#endif
+
 #include <errno.h>
 
-#include <rpc/xdr.h>
+#include <rpc/rpc.h>
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -91,7 +97,11 @@
 /** Indicates that the object was initialized from a file. */
 #define FILE_INIT  1
 /** Indicates that the object was initialized from a list. */
+#ifdef __APPLE__
+#define OLIST_INIT  2
+#else
 #define LIST_INIT  2
+#endif
 /** Indicates that the object was initialized directly. */
 #define DIRECT_INIT  3
 /** Time interval (in sec) at which job monitoring datagrams are sent. */
--- work/ApMon-CPP-2.0.5/ApMon.cpp	Wed Jul 13 11:44:30 2005
+++ work/ApMon-CPP-2.0.5/ApMon.cpp.new	Wed Jul 13 13:30:01 2005
@@ -149,7 +149,12 @@
   if (destinationsList == NULL) 
     throw runtime_error("[ constructFromList() ] Null destination list");
  
-  initType = LIST_INIT;
+#ifdef __APPLE__
+  initType = OLIST_INIT;
+#else
+   initType = LIST_INIT;
+#endif
+
   initMonitoring();
 
   pthread_mutex_init(&mutex, NULL);
@@ -424,7 +429,11 @@
 	strncpy(ifr.ifr_name, this -> interfaceNames[i], sizeof(ifr.ifr_name) - 1); 
 	if(ioctl(sockd, SIOCGIFADDR, &ifr)<0)
 	  continue;
-	memcpy(ip, ifr.ifr_hwaddr.sa_data+2, 4);
+#ifdef __APPLE__
+        memcpy(ip, ifr.ifr_addr.sa_data+2, 4);
+#else
+        memcpy(ip, ifr.ifr_hwaddr.sa_data+2, 4);
+#endif
 	strcpy(tmp_s, inet_ntoa(*(struct in_addr *)ip));
 	sprintf(logmsg, "Found local IP address: %s", tmp_s);
 	logger(FINE, logmsg);
