--- work/tokenauthz-1.0.13/Makefile.am.ori	2005-10-07 09:38:01.000000000 +0200
+++ work/tokenauthz-1.0.13/Makefile.am	2005-10-07 09:38:21.000000000 +0200
@@ -23,7 +23,7 @@
 
 bin_PROGRAMS = tokenauthztest
 tokenauthztest_SOURCES = tokenauthztest.cxx
-tokenauthztest_LDADD = -lpthread libTokenAuthz.la
+tokenauthztest_LDADD = -lcrypto -lpthread libTokenAuthz.la
 
 
 EXTRA_DIST = conf/TkAuthz.Authorization.allow-all \
