--- work/perl-ldap-0.4001/Makefile.PL.old	2010-12-20 11:24:32.000000000 +0100
+++ work/perl-ldap-0.4001/Makefile.PL	2010-12-20 11:24:43.000000000 +0100
@@ -26,13 +26,13 @@
 include 	('ExtUtils::AutoInstall');
 
 features(
-  'SASL authentication' => [
-    -default => 0,
-    'Authen::SASL'  => '2.00',
+  #'SASL authentication' => [
+#    -default => 0,
+#    'Authen::SASL'  => '2.00',
     #recommends('Digest::MD5'),		# for DIGEST-MD5 SASL mech
     #recommends('Digest::HMAC_MD5'),	# for CRAM-MD5 SASL mech
     #recommends('GSSAPI'),		# for GSSAPI SASL mech
-  ],
+ # ],
   'LDAP URLs' => [
     -default => 0,
     'URI::ldap' => 1.10,
