--- work/alien-common-v2-19/lib/AliEn/SE/Methods/torrent.pm.orig	2012-12-04 14:22:32.000000000 +0100
+++ work/alien-common-v2-19/lib/AliEn/SE/Methods/torrent.pm	2012-12-04 14:23:04.000000000 +0100
@@ -63,7 +63,7 @@
     }
   }
   $self->info("Finally, start the seeder");
-  system("aria2c $args --seed-time=10 --seed-ratio=0 -d $self->{LOCALFILE}_dir  $link > /dev/null 2&>1 &"); 
+  system("aria2c $args --seed-time=10 --seed-ratio=0 -d $self->{LOCALFILE}_dir  $link > /dev/null 2>&1 &"); 
   #"$PROGRAM" $ARGS --seed-time=10 --seed-ratio=0 "$@" &>/dev/null &
   $self->info("Returning");
   return $self->{LOCALFILE};
