--- work/uuid-1.6.2/Makefile.in.old	2010-08-17 16:23:44.000000000 +0200
+++ work/uuid-1.6.2/Makefile.in	2010-08-17 16:23:49.000000000 +0200
@@ -62,7 +62,7 @@
 PHP         = @PHP@
 PG_CONFIG   = @PG_CONFIG@
 
-LIB_NAME    = libuuid.la
+LIB_NAME    = libosspuuid.la
 LIB_OBJS    = uuid.lo uuid_md5.lo uuid_sha1.lo uuid_prng.lo uuid_mac.lo uuid_time.lo uuid_ui64.lo uuid_ui128.lo uuid_str.lo
 
 DCE_NAME    = libuuid_dce.la
