--- work/uuid-1.6.2/Makefile.in.old	2010-08-17 16:23:44.000000000 +0200
+++ work/uuid-1.6.2/Makefile.in	2010-08-17 16:23:49.000000000 +0200
@@ -62,7 +62,7 @@
 PHP         = @PHP@
 PG_CONFIG   = @PG_CONFIG@
 
-LIB_NAME    = libuuid.la
+LIB_NAME    = libosspuuid.la
 LIB_OBJS    = uuid.lo uuid_md5.lo uuid_sha1.lo uuid_prng.lo uuid_mac.lo uuid_time.lo uuid_ui64.lo uuid_ui128.lo uuid_str.lo
 
 DCE_NAME    = libuuid_dce.la
--- work/uuid-1.6.2/perl/Makefile.PL.old	2010-08-17 16:24:34.000000000 +0200
+++ work/uuid-1.6.2/perl/Makefile.PL	2010-08-17 16:24:48.000000000 +0200
@@ -33,8 +33,8 @@
 use ExtUtils::MakeMaker;
 
 #   determine source directory
-my ($srcdir) = map { my $d = $_; $d =~ s/\/libuuid\.la$//; $d }
-               grep { -f $_ } ("../libuuid.la", glob("../*/libuuid.la"))
+my ($srcdir) = map { my $d = $_; $d =~ s/\/libosspuuid\.la$//; $d }
+               grep { -f $_ } ("../libosspuuid.la", glob("../*/libosspuuid.la"))
     or die "no source directory found (where libuuid.la is located)";
 
 #   determine extra build options
@@ -47,7 +47,7 @@
     VERSION_FROM      => 'uuid.pm',
     ABSTRACT_FROM     => 'uuid.pod',
     PREREQ_PM         => {},
-    LIBS              => [ "-L$srcdir/.libs -L$srcdir -luuid" ],
+    LIBS              => [ "-L$srcdir/.libs -L$srcdir -losspuuid" ],
     DEFINE            => '',
     INC               => "-I. -I$srcdir",
     PM                => { 'uuid.pm'   => '$(INST_LIBDIR)/uuid.pm',
