--- work/openldap-2.4.23/servers/slapd/schema/core.schema.old	2010-10-04 09:31:11.000000000 +0200
+++ work/openldap-2.4.23/servers/slapd/schema/core.schema	2010-10-04 09:31:22.000000000 +0200
@@ -600,11 +600,3 @@
 	SUBSTR caseIgnoreIA5SubstringsMatch
 	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26 )
 
-# RFC 2459 -- deprecated in favor of 'mail' (in cosine.schema)
-attributetype ( 1.2.840.113549.1.9.1
-	NAME ( 'email' 'emailAddress' 'pkcs9email' )
-	DESC 'RFC3280: legacy attribute for email addresses in DNs'
-	EQUALITY caseIgnoreIA5Match
-	SUBSTR caseIgnoreIA5SubstringsMatch
-	SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{128} )
-
