--- ./work/openldap-2.0.27/configure.ori	2002-09-23 03:06:54.000000000 +0200
+++ ./work/openldap-2.0.27/configure	2005-09-23 14:06:54.000000000 +0200
@@ -607,7 +607,7 @@
   fi
 done
 
-
+CPPFLAGS="$CPPFLAGS -DBIND_8_COMPAT"
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
 ac_cpp='$CPP $CPPFLAGS'
@@ -17334,7 +17334,12 @@
     DYN_EXT=dll
 else
     PLAT=UNIX
-    DYN_EXT=so
+    case "$host_os" in
+	darwin*)
+	    DYN_EXT=dylib;;
+	*)
+	    DYN_EXT=so;;
+    esac
 fi
 
 
--- work/openldap-2.0.27/build/ltconfig.ori	2005-09-28 11:54:05.000000000 +0200
+++ work/openldap-2.0.27/build/ltconfig	2005-09-28 11:54:13.000000000 +0200
@@ -1381,11 +1381,11 @@
 
   darwin* | rhapsody*)
     allow_undefined_flag='-undefined warning'
-    archive_cmds='$CC $(if [ "$module" = "yes" ]; then echo -bundle; else echo -dynamiclib -flat_namespace -undefined suppress; fi) -o $lib $libobjs $deplibs $linkopts'
+    archive_cmds='$CC $(if [ "$module" = "yes" ]; then echo -bundle; else echo -dynamiclib -single_module -undefined dynamic_lookup -install_name $libdir/`basename $lib`; fi) -o $lib $libobjs $deplibs $linkopts'
     archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     ## What we need is to hardcode the path to the library, not the search path
-    #hardcode_direct=yes
-    #hardcode_libdir_flag_spec='-install_name $libdir/$lib'
+    hardcode_direct=yes
+    hardcode_libdir_flag_spec='-install_name $libdir/$lib'
     hardcode_shlibpath_var=no
     whole_archive_flag_spec='-all_load'
     ;;
