--- work/openldap-2.0.27/Portfile	2 Jan 2004 09:51:54 -0000	1.4
+++ work/openldap-2.0.27/Portfile	2 Jan 2004 10:48:42 -0000	1.5
@@ -1,4 +1,4 @@
-# $Id: Portfile.patch 25 2005-03-01 04:47:14Z buncic $
+# $Id: Portfile.patch 25 2005-03-01 04:47:14Z buncic $
 
 PortSystem 1.0
 name            openldap
@@ -28,7 +28,7 @@
		--localstatedir=${prefix}/var/run \
   
 platform darwin 7 {
-	patchfiles patch-dnssrv.c
+	patchfiles patch-dnssrv.c patch-Makefile.in
 }
   
 variant darwin {
  
 
