--- work/gsoap2.7-2.7/Makefile.am	2002-10-05 23:17:35.000000000 +0200
+++ work/gsoap2.7-2.6src/Makefile.am	2005-01-25 14:20:46.000000000 +0100
@@ -29,6 +29,9 @@
 	echo "| Thanks for using gsoap.                                |"; \
 	echo "|                                                        |"; \
 	echo "|               http://sourceforge.net/projects/gsoap2   |"; \
+	echo "|                                                        |"; \
+	echo "| Note: A set of stdsoap sources has been saved to       |"; \
+	echo "|       $(pkgdatadir) for building with different flags  |"; \
 	echo "+--------------------------------------------------------+"; 
 
 ## remove also the links for files like install.sh etc. 
--- work/gsoap2.7-2.7/soapcpp2/Makefile.am	2003-12-20 05:24:15.000000000 +0100
+++ work/gsoap2.7-2.6src/soapcpp2/Makefile.am	2005-01-25 14:17:32.000000000 +0100
@@ -35,9 +35,11 @@
 
 include_HEADERS = stdsoap2.h
 
+
 install-data-hook:
 	$(LN_S) $(top_builddir)/soapcpp2/src/soapcpp2 $(top_builddir)/soapcpp2/soapcpp2 || echo "ok, link already exists".
 	$(LN_S) $(top_builddir)/soapcpp2/wsdl/wsdl2h $(top_builddir)/soapcpp2/wsdl2h || echo "ok, link already exists".
-
+	mkdir -p $(pkgdatadir)
+	cp stdsoap2.h stdsoap2.c stdsoap2.cpp $(pkgdatadir)
 
 
