--- work/MIME-Lite-3.027/Makefile.PL.old	2009-09-17 16:13:24.000000000 +0200
+++ work/MIME-Lite-3.027/Makefile.PL	2010-08-10 14:09:08.000000000 +0200
@@ -20,16 +20,13 @@
 my %prereq = (
   'File::Spec'          => 0,
   'Email::Date::Format' => '1.000', # first release
+  'Mail::Address'     => 1.62,
+  'MIME::Types'       => 1.28,
+  'File::Basename'    => undef,
+  'MIME::Base64'      => undef,
+  'MIME::QuotedPrint' => undef,
+
 );
-if (prompt("Add prereqs?",'Yes')=~/y/i) {
-    %prereq=( %prereq,
-              'Mail::Address'     => 1.62,
-              'MIME::Types'       => 1.28,
-              'File::Basename'    => undef,
-              'MIME::Base64'      => undef,
-              'MIME::QuotedPrint' => undef,
-            );
-}
 
 # Write the Makefile:
 WriteMakefile(
