--- work/mysql-5.0.26/configure	2006-07-27 20:53:44.000000000 +0200
+++ work/mysql-5.0.26/configure.new	2006-08-16 15:10:17.000000000 +0200
@@ -39038,8 +39038,8 @@
         OPENSSL_INCLUDE=-I$incs
       fi
       # Test for libssl using all known library file endings
-      if test -f $d/libssl.a || test -f $d/libssl.so || \
-         test -f $d/libssl.sl || test -f $d/libssl.dylib ; then
+      if test -f $libs/libssl.a || test -f $libs/libssl.so || \
+         test -f $libs/libssl.sl || test -f $libs/libssl.dylib ; then
         OPENSSL_LIB=$libs
       fi
       ;;
