--- work/mysql-5.1.61/cmd-line-utils/libedit/term.c.old	2010-10-19 17:20:44.000000000 +0200
+++ work/mysql-5.1.61/cmd-line-utils/libedit/term.c	2010-10-19 17:20:49.000000000 +0200
@@ -901,7 +901,7 @@
 term_set(EditLine *el, const char *term)
 {
 	int i;
-	char buf[TC_BUFSIZE];
+//	char buf[TC_BUFSIZE];
 	char *area;
 	const struct termcapstr *t;
 	sigset_t oset, nset;
@@ -911,7 +911,7 @@
 	(void) sigaddset(&nset, SIGWINCH);
 	(void) sigprocmask(SIG_BLOCK, &nset, &oset);
 
-	area = buf;
+	area = NULL;
 
 
 	if (term == NULL)
