--- work/gt4.2.1-all-source-installer/source-trees-thr/database/c/sqliteodbc/sqliteodbc-0.74/configure	2010-03-08 21:22:17.000000000 +0100
+++ work/gt4.2.1-all-source-installer/source-trees-thr/database/c/sqliteodbc/sqliteodbc-0.74/configure.new	2010-03-16 19:14:17.000000000 +0100
@@ -1403,7 +1403,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
+  lt_cv_deplibs_check_method='file_magic Mach-O 64-bit dynamically linked shared library x86_64'
   lt_cv_file_magic_cmd='/usr/bin/file -L'
   case "$host_os" in
   rhapsody* | darwin1.012)
